### PR TITLE
feat: add tags to messages with contracts deployments

### PR DESCRIPTION
**Pull request by ppedziwiatr**

### CODE REVIEW

File: src/gateway/tasks/contractsMetadata.ts [hunk 1]
```diff
@@ -240,13 +240,8 @@ async function loadContractsMetadata(context: GatewayContext) {
...
         ]);
 
-      sleep(2000)
-        .then(() => {
-          sendNotificationToCache(context, definition.txId, definition.initState);
-        })
-        .catch((e) => {
-          logger.error(`No sleep 'till Brooklyn.`, e);
-        });
+      // TODO: add tags to ContractDefinition type in the SDK
```
Comment (ppedziwiatr) [inline src/gateway/tasks/contractsMetadata.ts:243]: Currently the tags won't be set for the contracts deployed directly to Arweave.

File: src/gateway/router/routes/sequencerRoute.ts [hunk 2]
```diff
@@ -166,7 +166,7 @@ export async function sequencerRoute(ctx: Router.RouterContext) {
     await trx.commit();
     sLogger.info('Total sequencer processing', benchmark.elapsed());
 
-    sendNotificationToCache(ctx, contractTag, undefined, interaction);
+    sendNotification(ctx, contractTag, undefined, interaction);
```
Comment (asiaziola) [inline src/gateway/router/routes/sequencerRoute.ts:169]: does it stay undefined on purpose? the tags for interactions are sent differently?

Comment (ppedziwiatr) [reply src/gateway/router/routes/sequencerRoute.ts:169]: the interaction itself contains the tags

**Comment by ppedziwiatr** [comment]
> The only problem I see with this one is that tags will be passed in the encoded form but I believe it's a matter of encoding tags passed in the aggregate node

are you sure? I was pretty sure that it's the opposite and that they will be in a decoded form?:-)

File: src/gateway/router/routes/deployContractRoute.ts [hunk 3]
```diff
@@ -137,13 +137,7 @@ export async function deployContractRoute(ctx: Router.RouterContext) {
...
     }
 
-    sleep(2000)
-      .then(() => {
-        sendNotificationToCache(ctx, contractTx.id, initState);
-      })
-      .catch((e) => {
-        logger.error(`No sleep 'till Brooklyn.`, e);
-      });
+    sendNotification(ctx, contractTx.id, {initState, tags: contractTags});
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/deployContractRoute.ts:140]: I'm pretty sure that `contractTags` are decoded - they come from the `prepareTags` function, which handles the decoding

File: src/gateway/router/routes/deployBundledRoute.ts [hunk 4]
```diff
@@ -69,14 +68,7 @@ export async function deployBundledRoute(ctx: Router.RouterContext) {
...
     await gatewayDb('contracts').insert(insert);
-
-    sleep(2000)
-      .then(() => {
-        sendNotificationToCache(ctx, bundlrResponse.data.id, initState);
-      })
-      .catch((e) => {
-        logger.error(`No sleep 'till Brooklyn.`, e);
-      });
+    sendNotification(ctx, bundlrResponse.data.id, {initState, tags: dataItem.tags});
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/deployBundledRoute.ts:71]: I believe that dataItem's tags are also already decoded - they are used in the `verifyContractTags` without additional decoding

File: src/gateway/router/routes/deployContractRoute_v2.ts [hunk 5]
```diff
@@ -121,13 +121,7 @@ export async function deployContractRoute_v2(ctx: Router.RouterContext) {
...
     await gatewayDb('contracts').insert(insert);
 
-    sleep(2000)
-      .then(() => {
-        sendNotificationToCache(ctx, bundlrResponse.data.id, initState);
-      })
-      .catch((e) => {
-        logger.error(`No sleep 'till Brooklyn.`, e);
-      });
+    sendNotification(ctx, bundlrResponse.data.id, {initState, tags: contractDataItem.tags});
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/deployContractRoute_v2.ts:124]: again - tags in DateItem are IMO decoded

File: src/gateway/router/routes/registerContractRoute.ts [hunk 6]
```diff
@@ -103,13 +103,7 @@ export async function registerContractRoute(ctx: Router.RouterContext) {
...
     await gatewayDb('contracts').insert(insert);
 
-    sleep(2000)
-      .then(() => {
-        sendNotificationToCache(ctx, txId, initState);
-      })
-      .catch((e) => {
-        logger.error(`No sleep 'till Brooklyn.`, e);
-      });
+    sendNotification(ctx, txId, {initState, tags});
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/registerContractRoute.ts:106]: tags come from verifyContractTags - so also IMO already decoded.

**Pull request merged**